src/jit/hwintrinsicArm64.cpp

```diff
@@ -182,10 +202,36 @@ GenTree* Compiler::impHWIntrinsic(NamedIntrinsic        intrinsic,
 
             return gtNewSimdHWIntrinsicNode(simdType, op1, op2, intrinsic, simdBaseType, simdSizeBytes);
 
+        case HWIntrinsicInfo::SimdSelectOp:
+            // op1 is the first operand
+            // op2 is the second operand
+            // op3 is the third operand
+            op3 = impSIMDPopStack(simdType);
```

Comment:
It might be good to have a general helper method for popping/validating the types, as was requested/done for x86.

Reply:
I am not sure why.  Each form has different requirements.  Not sure how a helper would help.

Maybe `impScalarPopStack()`

Reply:
x86 has this method: https://github.com/dotnet/coreclr/blob/master/src/jit/hwintrinsicxarch.cpp#L276

Which validates the type of a struct is a SIMD Type and the type of a scalar matches what the signature expects.

Reply:
I believe @CarolEidt was the one who requested we validate the type of the scalar values as well, rather than just calling `impPopStack().val`

Reply:
I was assuming we relied on the C# (et al) compiler to validate this.  
Perhaps this is to check hand written IL.
Or perhaps it is just defensive programming,

In any case I think this can safely be a separate PR.

Reply:
> Or perhaps it is just defensive programming,

Yes, that's the idea (that's true of many of the asserts and checks in the JIT, but it's surprising how often the "obvious" checks catch a problem.

> In any case I think this can safely be a separate PR.

Me too